src/main/java/seedu/address/logic/commands/editcommand/EditAssignmentCommand.java

```diff
@@ -0,0 +1,102 @@
+package seedu.address.logic.commands.editcommand;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ASSIGNMENT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE;
+
+import java.time.LocalDateTime;
+
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.module.Assignment;
+import seedu.address.model.module.Description;
+import seedu.address.model.module.Module;
+
+public class EditAssignmentCommand extends EditCommand {
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits a module in RemindMe."
+            + "Parameters: "
+            + PREFIX_MODULE + " MODULE TITLE "
+            + PREFIX_ASSIGNMENT + " ASSIGNMENT INDEX "
+            + "[" + PREFIX_DESCRIPTION + " NEW DESCRIPTION OR"
+            + PREFIX_DEADLINE + " NEW DEADLINE]"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_MODULE + "CS2103T"
+            + PREFIX_ASSIGNMENT + "1"
+            + PREFIX_DESCRIPTION + "Complete DG";
+
+    public static final String MESSAGE_SUCCESS = "Assignment edited: %1$s";
+    public static final String MESSAGE_NO_MODULE = "This module does not exists in RemindMe";
+    public static final String MESSAGE_NO_ASSIGNMENT = "This module does not contain any assignment at this index.";
+    public static final String MESSAGE_NO_CHANGE = "The input given does not change anything!";
+    public static final String MESSAGE_TWO_CHANGES = "Only one field can be changed at a time.";
+    public static final String MESSAGE_NO_VALID_CHANGES = "Please input a valid edit.";
+
+    private final Module module;
+    private final int toEdit;
```

Comment:
maybe using toEditIndex makes its purpose more obv?

Reply:
thanks for pointing that out!